checks/permissions.go

```diff
@@ -39,15 +39,16 @@ func leastPrivilegedTokens(c *checker.CheckRequest) checker.CheckResult {
 
 func validatePermission(key string, value interface{}, path string,
 	logf func(s string, f ...interface{})) (bool, error) {
-	switch val := value.(type) {
-	case string:
-		if val == "write" {
-			logf("!! token-permissions/github-token - %v permission set to '%v' in %v", key, val, path)
-			return false, nil
-		}
-	default:
+	val, ok := value.(string)
+	if !ok {
 		return false, ErrInvalidGitHubWorkflowFile
 	}
+
+	if val == "write" {
```

Comment:
instead of the direct comparison can we do `strings.EqualFold`? which is case insensitive

Reply:
done. Did the same in other comparison places too. I don't have unit tests for this though. I actually don't know if github enforces lower-case or not.